cmd/ci-operator/main.go

```diff
@@ -243,6 +258,38 @@ func main() {
 	opt.Report()
 }
 
+// setupLogger sets up logrus to print all logs to a file and user-friendly logs to stdout
+func setupLogger() (*secrets.DynamicCensor, io.Closer, error) {
+	logrusutil.ComponentInit()
+	logrus.SetLevel(logrus.TraceLevel)
+	censor := secrets.NewDynamicCensor()
+	logrus.SetFormatter(logrusutil.NewFormatterWithCensor(logrus.StandardLogger().Formatter, &censor))
+	logrus.SetOutput(ioutil.Discard)
+	logrus.AddHook(&writer.Hook{
+		Writer: os.Stdout,
+		LogLevels: []logrus.Level{
+			logrus.InfoLevel,
+			logrus.WarnLevel,
+			logrus.ErrorLevel,
+			logrus.FatalLevel,
+			logrus.PanicLevel,
+		},
+	})
+	artifactDir, set := api.Artifacts()
+	if !set {
+		return &censor, nil, nil
+	}
+	verboseFile, err := os.Open(filepath.Join(artifactDir, "ci-operator.log"))
```

Comment:
Why, doesn't Prow give us this already?

Reply:
I don't think Prow has any "user-facing" components like `ci-operator`. Perhaps there's something like this in `kubetest`?